### PR TITLE
tests: add more GCE tests

### PR DESCRIPTION
**Pull request by georgiyekkert**

### CODE REVIEW

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs [hunk 1]
```diff
@@ -122,4 +122,248 @@ private Operation PollForCompletion(Operation operation, string alias)
             return operation;
         }
     }
+    [Collection(nameof(ComputeFixture))]
+    public sealed class InstancesTests{
```
Comment (jskeet) [inline apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs:126]: Please use brace-on-new-line formatting - see the existing code for examples.

Comment (jskeet) [reply apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs:126]: (If you're using an IDE, I'd expect that just autoformatting the whole document should be fine.)

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs [hunk 2]
```diff
@@ -122,4 +122,248 @@ private Operation PollForCompletion(Operation operation, string alias)
             return operation;
         }
     }
+    [Collection(nameof(ComputeFixture))]
```
Comment (jskeet) [inline apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs:125]: Please add a blank line between methods.

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs [hunk 3]
```diff
@@ -122,4 +122,248 @@ private Operation PollForCompletion(Operation operation, string alias)
...
+            PatchInstance();
+            DeleteInstance();
+
+            void CreateInstanceNonAsciiDescription(){
+                AttachedDiskInitializeParams init_params = new AttachedDiskInitializeParams
+                {
+                    SourceImage = diskImage
+                };
+                Instance instanceResource = new Instance
+                    {
```
Comment (jskeet) [inline apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs:153]: Outdent so the brace is under the "I" of "Instance"

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs [hunk 4]
```diff
@@ -122,4 +122,248 @@ private Operation PollForCompletion(Operation operation, string alias)
...
+            InstancesClient instancesClient = InstancesClient.Create();
+            var instanceName = _fixture.GenerateResourceId();
+
+            CreateInstanceNonAsciiDescription();
+            FetchInstance();
+            PatchInstance();
+            DeleteInstance();
+
+            void CreateInstanceNonAsciiDescription(){
+                AttachedDiskInitializeParams init_params = new AttachedDiskInitializeParams
```
Comment (jskeet) [inline apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs:148]: Nit: init_params => initParams - or better yet, just inline this into AttachedDisk object initializer.

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs [hunk 5]
```diff
@@ -122,4 +122,248 @@ private Operation PollForCompletion(Operation operation, string alias)
...
+        }
+        [Fact (Skip="resize fails")]
+        public void TestQueryParams(){
+            string projectId = _fixture.ProjectId;
+            string zone = _fixture.Zone;
+            InstanceGroupManagersClient instanceGroupManagersClient = InstanceGroupManagersClient.Create();
+            InstanceTemplatesClient instanceTemplatesClient = InstanceTemplatesClient.Create();
+            var instanceGroupManagerName = _fixture.GenerateResourceId();
+            var instanceTemplateName = _fixture.GenerateResourceId();
+            AttachedDiskInitializeParams init_params = new AttachedDiskInitializeParams
```
Comment (jskeet) [inline apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs:219]: Same comment as earlier - I'd inline this.

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs [hunk 6]
```diff
@@ -122,4 +122,248 @@ private Operation PollForCompletion(Operation operation, string alias)
...
+            InstanceProperties props = new InstanceProperties{
+                Disks = {
+                            new AttachedDisk{
+                                AutoDelete = true,
+                                Boot = true,
+                                Type = AttachedDisk.Types.Type.Persistent,
+                                InitializeParams = init_params
+                            }
+                        },
+                NetworkInterfaces = {
```
Comment (jskeet) [inline apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs:232]: I think the whole of the NetworkInterfaces initializer can be on one line. (But space around the "=")

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs [hunk 7]
```diff
@@ -122,4 +122,248 @@ private Operation PollForCompletion(Operation operation, string alias)
...
+                Operation resizeOp = instanceGroupManagersClient.Resize(
+                    projectId,
+                    zone,
+                    instanceGroupManagerName,
+                    0
+                );
+
+                PollForCompletion(resizeOp, "resize");
+
+                InstanceGroupManager instanceGroupManagerResized = instanceGroupManagersClient.Get(
```
Comment (jskeet) [inline apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs:274]: I'd probably use `var` and a shorter variable name (maybe just `fetched`?) and put the whole method call on one line.

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs (hunk 1 above)
Comment (jskeet) [inline apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs:126]: This should be in a new file called InstancesTest.cs (and called InstancesTest rather than InstancesTests). Alternatively, just leave it as part of AcceptanceTest if we don't particularly need it to be broken out.

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs [hunk 8]
```diff
@@ -122,4 +122,248 @@ private Operation PollForCompletion(Operation operation, string alias)
...
+                    instanceName,
+                    shieldedInstanceConfigResource);
+                PollForCompletion(patchOp, "patch");
+                Instance instance = instancesClient.Get(projectId, zone, instanceName);
+                Assert.Equal(true, instance.ShieldedInstanceConfig.EnableSecureBoot);
+            }
+
+        }
+        [Fact (Skip="resize fails")]
+        public void TestQueryParams(){
```
Comment (jskeet) [inline apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs:212]: It's not clear why we'd need a second test for whether query parameters are transcoded properly or not.

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs [hunk 9]
```diff
@@ -122,4 +122,248 @@ private Operation PollForCompletion(Operation operation, string alias)
...
+                    projectId,
+                    instanceTemplateName
+                );
+
+                PollForCompletionGlobalOp(deleteTemplateOp, "delete template");
+            }
+
+        }
+
+        private Operation PollForCompletion(Operation operation, string alias)
```
Comment (jskeet) [inline apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs:301]: I really hope we don't need this code (and PollForCompletionGlobalOp) in multiple places.

**Review by jskeet — COMMENTED** [review]
Mostly formatting, but a few other aspects in terms of where we have the tests etc.

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs (hunk 3 above)
Comment (georgiyekkert) [reply apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs]: Done

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs (hunk 5 above)
Comment (georgiyekkert) [reply apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs]: Done

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs (hunk 6 above)
Comment (georgiyekkert) [reply apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs]: Done

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs (hunk 1 above)
Comment (georgiyekkert) [reply apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs]: Moved

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs (hunk 8 above)
Comment (georgiyekkert) [reply apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs]: This is an edge case(a bug was found during testing for other client libraries) - we want to test that 0 won't be omitted in query parameters.

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs (hunk 1 above)
Comment (georgiyekkert) [reply apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs]: done. autoformatted via Resharper CLI tools.

**Comment by georgiyekkert** [comment]
@viacheslav-rostovtsev please take a look

**Comment by jskeet** [comment]
(We can ignore the diff check; I'm not quite sure why it's triggered here, but that's fine.)

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs [hunk 10]
```diff
@@ -21,13 +21,13 @@
 namespace Google.Cloud.Compute.V1.IntegrationTests
 {
     /// <summary>
-    /// Miscellaneous acceptance tests for the Compute API.
+    ///     Miscellaneous acceptance tests for the Compute API.
```
Comment (jskeet) [inline apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs:24]: Please unindent this. (It may well be ReSharper indenting by default, but it's harder to read like this.)

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/ComputeFixture.cs [hunk 11]
```diff
@@ -27,51 +27,70 @@ public class ComputeFixture : CloudProjectFixtureBase, ICollectionFixture<Comput
...
+                throw new InvalidOperationException(
+                    $"Environment variable {TestRegionEnvironmentVariable} must be set");
             }
-            ResourcePrefix = IdGenerator.FromDateTime(prefix: "test-", suffix: "-");
+
+            ResourcePrefix = IdGenerator.FromDateTime("test-", "-");
         }
 
-        public string GenerateResourceId() => IdGenerator.FromGuid(prefix: ResourcePrefix);
+        /// <summary>
```
Comment (jskeet) [inline apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/ComputeFixture.cs:43]: Why have you moved these? We generally keep properties before constructors. If this is Resharper reformatting, please just revert it all - it's made a lot of changes, almost all of which have negative results on readability.

**Review by jskeet — COMMENTED** [review]
I won't review the rest yet, as there are a *lot* of formatting changes here which are harmful. Please revert those, and then it'll be much easier to see the real changes.

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs (hunk 10 above)
Comment (georgiyekkert) [reply apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/AcceptanceTest.cs]: reverted

File: apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/ComputeFixture.cs (hunk 11 above)
Comment (georgiyekkert) [reply apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.IntegrationTests/ComputeFixture.cs]: These were moved by resharper, I haven't found editorconfig file in this repo so I used the one from dotnet/corefx repo. 
I reverted all the changes made by resharper and fixed your comments manually

**Review by jskeet — APPROVED** [review]
Thanks for all of this. You're right, we don't have an editorconfig - we've typically found that automatic formatting ends up being less readable than just being careful, and there are few enough contributors that we've generally been able to keep things consistent.

I'm going to merge this now, but then have a look and see if I can refactor the polling a bit (which will be easier to examine in VS - it may be there's nothing to do yet).

**Comment by jskeet** [comment]
I'm not sure what's going on with all the GitHub actions here... will investigate and try to fix it myself.

**Comment by jskeet** [comment]
Humbug. I've force pushed to your master slightly inaccurately, which has closed this PR. Unfortunately that means I can no longer force push the *correct* commit to your master. I'll recreate this PR from my own fork.
